src/app/(components)/common/text-input.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ReactElement } from "react";
+import { classNames } from "~/utils/classnames";
+
+export type InputProps = React.DetailedHTMLProps<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  HTMLInputElement
+>;
+
+export const TextInput = React.forwardRef<
+  HTMLInputElement,
+  {
+    prefixEl?: ReactElement | string;
+    suffixEl?: ReactElement | string;
+    className?: string;
+  } & InputProps
+>((props, ref) => {
+  const { prefixEl, suffixEl, className, ...rest } = props;
+  return (
+    <div
+      className={classNames(
+        "flex rounded-md border border-gray-700 px-1 shadow-sm focus-within:border-pink-500 focus-within:ring-1 focus-within:ring-pink-500",
+        className
+      )}
+    >
+      {prefixEl && (
```

Comment:
This can pass as a falsy string and still render, i.e. the number 0. Throw a !! in front